File: pyjl/tests_new/scope_test.py
def func():
    return "aaaa"

class TestClass:
    def func(self):
        return "ssss"

def test():
    num:int = 2
    teststr:str = "ola"
    def inner_test():
        print(num*teststr)
    
    def inner_test_2():
        num: int = 4
        print(num*teststr)
    
    inner_test()
    inner_test_2()

if __name__ == "__main__":
    print(func())
    testClass = TestClass()
    print(testClass.func())

    # Function scopes
    test()

File: pyjl/tests_new/scope_test.jl
function func()::String
return "aaaa"
end

struct TestClass
end
function func(self::TestClass)::String
return "ssss"
end


function test()
num::Int64 = 2
teststr::String = "ola"
function inner_test()
println(repeat(teststr,num));
end

function inner_test_2()
num::Int64 = 4
println(repeat(teststr,num));
end

inner_test();
inner_test_2();
end

function main()
println(func());
testClass = TestClass()
println(func(testClass));
test();
end

main()
